src/rt-app_taskgroups.c

```diff
@@ -0,0 +1,354 @@
+#include <stdlib.h>
+#include <string.h>
+#include <mntent.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+
+#include "config.h"
+#include "rt-app_utils.h"
+#include "rt-app_taskgroups.h"
+
+static taskgroup_ctrl_t tg_ctrl;
+
+static int cgroup_check_cpu_controller(void)
+{
+	int dummy[2], enabled, ret = 0;
+	FILE *cgroups;
+	char buf[512];
+
+	cgroups = fopen("/proc/cgroups", "re");
+
+	if (!cgroups) {
+		perror("fopen");
+		goto err;
+	}
+
+	/* Ignore the first line as it contains the header. */
+	if (!fgets(buf, sizeof(buf), cgroups)) {
+		perror("fgets");
+		goto err;
+	}
+
+	while (!feof(cgroups)) {
+		/*
+		 * Only interested in 'subsys_name' and 'enabled' column
+		 * of /proc/cgroups, not in 'hierarchy' or 'num_cgroups'.
+		 */
+		if (fscanf(cgroups, "%s %d %d %d", buf, &dummy[0], &dummy[1],
+			   &enabled) < 4) {
+			perror("fscanf");
+			goto err;
+		}
+
+		if (!strcmp(buf, "cpu") && enabled == 1)
+			goto done;
+	}
+err:
+	ret = -1;
+done:
+	if (cgroups)
+		fclose(cgroups);
+
+	return ret;
+}
+
+static int cgroup_get_cpu_controller_mount_point(void)
+{
+	struct mntent *ent;
+	FILE *mounts;
+	int ret = -1;
+
+	mounts = fopen("/proc/mounts", "re");
```

Comment:
this is supposed to use setmntent() and endmntent() rather than fopen/fclose (according to the man page - I suppose it might work on one C library and not others if we don't follow the rules)

Reply:
I will change this.